daemon/daemon_unix.go

```diff
@@ -653,6 +653,9 @@ func configureKernelSecuritySupport(config *Config, driverName string) error {
 }
 
 func (daemon *Daemon) initNetworkController(config *Config, activeSandboxes map[string]interface{}) (libnetwork.NetworkController, error) {
+	if err := processPoolsConfig(config); err != nil {
+		return nil, err
+	}
```

Comment:
`I do not think we will allow the pools to be modified on reload.`

I was referring to the case where the daemon is stopped and restarted with a new range. For modification through the config file, agree that we don't have to support it.

From an operator pov they want to configure the pool to avoid conflicts with the underlay networks or the host subnets.. so after configuring a range, the only reason one would modify the range is because some subnets in the range can no longer be used by docker.. so the user should have deleted the old networks before making this config change.. if he doesn’t its a user’s mistake..But unless its absolutely impossible for us to detect that misconfig and error out we should do it.. allowing that erroneous config can lead to hard to debug issues.

For the swarm mode we have to think about what is the best way to support this config. Since the `swarm init` is done independent of the daemon configs its more natural to associate the global range config with `swarm init` rather than the engine config. But it should be possible to do the validity checks in that case also because swarm mode will deal only with global range and the engine will only deal with local range.